docs/error-codes.md

```diff
@@ -1569,6 +1569,10 @@ void TestMethod(MethodInfo methodInfo)
 }
 ```
 
+#### `IL2104`: Assembly 'assembly' produced trim warnings
+
+The assembly 'assembly' produced trim analysis warnings in the context of the app. This means the assembly has not been fully annotated for trimming. Consider contacting the library author to request they add trim annotations to the library. To see detailed warnings for this assembly, turn off warning collapsing by passing either `--collapse-` to show detailed warnings for all assemblies, or `--collapse- "assembly"` to show detailed warnings for that assembly.
```

Comment:
Seems long. I wonder if we can shorten it without distorting the message too much.

> The assembly 'assembly' is not trim-safe and may reference members that are trimmed away in the final app. Consider contacting the library author to request they add trim annotations to the library. For more information, see \<link to docs>

I'm not sure seeing the detailed warnings is that helpful, so I would suggest putting that in the docs, along with instructions for the library author who may get this info after the user contacts them.

Reply:
My mistake, I think I read this as the warning message, when it's just the warning description.

In that case, I think the message `Assembly 'assembly' produced trim warnings` is too brief. We should either include a link for more information, or include a recommendation to contact the author in the message directly.